store/src/java/com/zimbra/cs/imap/RemoteImapMailboxStore.java

```diff
@@ -107,7 +101,10 @@ public void deleteMessages(OperationContext octxt, List<Integer> ids) {
         }
     }
 
-    /** @return List of IMAP UIDs */
+    /**
+     * MUST only be called when the source items and target folder are in the same mailbox
+     * @return List of IMAP UIDs
+     */
     @Override
```

Comment:
Would it be smart to add a runtime check to validate this requirement such that it fails nicely?
(Perhaps it's done inside zMailbox.imapCopy ? )

Reply:
Well, in some senses the requirements will always be met - because the API only talks about IDs in terms of a single mailbox.  The comment is really a warning about that limitation - if it were to be called in the wrong circumstances, odd things would happen.  The only "fails nicely" bit would be if some of the IDs from the wrong mailbox happened not to exist in the one being used then there would be failures.

Reply:
Thanks for the explanation 👍